File: android/src/main/kotlin/com/cleveradssolutions/plugin/flutter/FlutterAd.kt
package com.cleveradssolutions.plugin.flutter

import android.app.Activity
import com.cleveradssolutions.sdk.AdContentInfo
import io.flutter.plugin.platform.PlatformView

internal interface FlutterAd {
    val adId: Int
    val contentInfo: AdContentInfo?
    var isAutoloadEnabled: Boolean
    var isAutoshowEnabled: Boolean
    var interval: Int

    fun isLoaded(): Boolean

    fun load()

    fun showScreen(activity: Activity?)

    /**
     * Gets the PlatformView for the ad. Default behavior is to return null. Should be overridden by
     * ads with platform views, such as banner and native ads.
     */
    fun getPlatformView(width: Int, height: Int): PlatformView?

    /**
     * Invoked when dispose() is called on the corresponding Flutter ad object. This perform any
     * necessary cleanup.
     */
    fun dispose()
}

File: ios/Classes/FlutterAd.swift
import CleverAdsSolutions

protocol FlutterAd: AnyObject {
    var adId: Int { get }

    var contentInfo: AdContentInfo? { get }

    var isAutoloadEnabled: Bool { get set }
    var isAutoshowEnabled: Bool { get set }
    var interval: Int { get set }

    func isAdLoaded() -> Bool

    func loadAd()

    /**
     * Invoked when dispose() is called on the corresponding Flutter ad object. This perform any
     * necessary cleanup.
     */
    func disposeAd()
}
